wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -922,6 +931,29 @@ private MethodSpec messageFieldsAndUnknownFieldsConstructor(
     return result.build();
   }
 
+  /** The private no-arg constructor is used when an object is deserialized. */
+  // Example:
+  //
+  // private SimpleMessage() {
+  //   this(null, Internal.newMutableList(), ByteString.EMPTY);
+  // }
+  //
+  private MethodSpec messageNoArgsConstructor(MessageType type) {
+    MethodSpec.Builder result = MethodSpec.constructorBuilder().addModifiers(PRIVATE);
+    result.addJavadoc("Used for deserialization.\n");
```

Comment:
This comment is misleading. The constructor is not used for proto deserialization but only JSON deserialization via Gson and only when you are not using the supplied type adapter.